e2e/cypress/integration/messaging/permalink_message_edit_spec.js

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// [number] indicates a test step (e.g. # Go to a page)
+// [*] indicates an assertion (e.g. * Check the title)
+// Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Permalink message edit', () => {
+    it('M18717 - Edit a message in permalink view', () => {
+        // # Login as "user-1" and go to /
+        cy.apiLogin('user-1');
+        cy.visit('/');
+
+        const searchWord = `searchtest ${Date.now()}`;
+
+        // # Post message
+        cy.postMessage(searchWord);
+
+        // # Search for searchWord
+        cy.get('#searchBox').type(searchWord).type('{enter}').then(() => {
```

Comment:
Sorry, but one more possible nit pick. I don't think you need to do a `.then` here, unless I'm missing something. Same for the `.then` on line 49.

Reply:
Unnecessary promise has been removed.  I was having difficulty selecting the first search result initially, and I thought it might be an issue of timing that a  `.then` would perhaps solve.  After I successfully targeted the selector, I should have checked to see if the promise was necessary.  

Thank you for your feedback.  Changes have been submitted.